plugins/woocommerce/src/Admin/Overrides/OrderTraits.php

```diff
@@ -42,29 +44,35 @@ public function get_item_shipping_amount( $item ) {
 	 * @todo If WC is currently not tax enabled, but it was before (or vice versa), would this work correctly?
 	 *
 	 * @param WC_Order_Item $item Line item from order.
+	 * @param int           $order_items_count   (optional) The number of order items in an order. This could be the remaining items left to refund.
+	 * @param float         $shipping_tax_amount (optional) The shipping tax amount in an order. This could be the remaining shipping tax amount left to refund.
 	 *
 	 * @return float|int
 	 */
-	public function get_item_shipping_tax_amount( $item ) {
-		$order_items = $this->get_item_count();
+	public function get_item_shipping_tax_amount( $item, $order_items_count = null, $shipping_tax_amount = null ) {
+		$order_items = $order_items_count ? $order_items_count : $this->get_item_count();
 		if ( 0 === $order_items ) {
 			return 0;
 		}
 
 		$product_qty               = $item->get_quantity( 'edit' );
 		$order_taxes               = $this->get_taxes();
 		$line_items_shipping       = $this->get_items( 'shipping' );
-		$total_shipping_tax_amount = 0;
-		foreach ( $line_items_shipping as $item_id => $shipping_item ) {
-			$tax_data = $shipping_item->get_taxes();
-			if ( $tax_data ) {
-				foreach ( $order_taxes as $tax_item ) {
-					$tax_item_id                = $tax_item->get_rate_id();
-					$tax_item_total             = isset( $tax_data['total'][ $tax_item_id ] ) ? (float) $tax_data['total'][ $tax_item_id ] : 0;
-					$total_shipping_tax_amount += $tax_item_total;
+		$total_shipping_tax_amount = $shipping_tax_amount ? $shipping_tax_amount : 0;
+
+		if ( 0 === $total_shipping_tax_amount ) {
```

Comment:
I know we check against 0 before we call the function. But wouldn't it be safer to check the passed shipping_tax_amount to stay closer to the original implmentation?

```suggestion
		if ( null === $shipping_tax_amount ) {
```